src/client/pfs.go

```diff
@@ -826,36 +864,54 @@ func (c APIClient) newPutObjectWriteCloser(tags ...string) (*putObjectWriteClose
 		request: &pfs.PutObjectRequest{
 			Tags: _tags,
 		},
-		putObjectClient: putObjectClient,
+		client: client,
 	}, nil
 }
 
 func (w *putObjectWriteCloser) Write(p []byte) (int, error) {
-	bytesWritten := 0
-	for {
-		// Buffer the write so that we don't exceed the grpc
-		// MaxMsgSize. This value includes the whole payload
-		// including headers, so we're conservative and halve it
-		ceil := bytesWritten + grpcutil.MaxMsgSize/2
-		if ceil > len(p) {
-			ceil = len(p)
-		}
-		actualP := p[bytesWritten:ceil]
-		if len(actualP) == 0 {
-			break
-		}
-		w.request.Value = actualP
-		if err := w.putObjectClient.Send(w.request); err != nil {
-			return 0, grpcutil.ScrubGRPC(err)
-		}
-		w.request.Value = nil
-		bytesWritten += len(actualP)
+	w.request.Value = p
+	if err := w.client.Send(w.request); err != nil {
+		return 0, grpcutil.ScrubGRPC(err)
```

Comment:
Is there a reason we no longer need the splitting code here? Or is it happening somewhere else?

Reply:
Yeah, the `ReadWrapper` above, combined with `io.CopyBuffer`, ensures that we will always split messages into chunks of size `len(buffer)`.